### PR TITLE
Carousel slider

### PR DESCRIPTION
**Pull request by OleanjiKingCode**
# Overflow of Carousel on mobile

## How should this be tested?

Before 
image

Now
image

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/1224

### CODE REVIEW

**Review by Adebesin-Cell — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/66301634/233379055-ef299264-5c36-4911-88cb-cf87777e21c8.mov

I think we can have this feature slider once the wikis promoted are more than 5 or 4, it'll make it look very nice and compact like in the design.

**Comment by OleanjiKingCode** [comment]
> Screen.Recording.2023-04-20.at.2.12.07.PM.mov 
> I think we can have this feature slider once the wikis promoted are more than 5 or 4, it'll make it look very nice and compact like in the design.

what was used to make this carousel

**Comment by Adebesin-Cell** [comment]
> > Screen.Recording.2023-04-20.at.2.12.07.PM.mov
> > I think we can have this feature slider once the wikis promoted are more than 5 or 4, it'll make it look very nice and compact like in the design.
> 
> what was used to make this carousel

Let me check 😅, I didn't check.

File: src/components/Elements/Carousel/Carousel.tsx [hunk 1]
```diff
@@ -1,16 +1,25 @@
...
 import { IconType } from 'react-icons/lib'
 import 'slick-carousel/slick/slick.css'
 import 'slick-carousel/slick/slick-theme.css'
 
-const Slider = SliderClass as unknown as (props: {
-  children: React.ReactNode
-  nextArrow: JSX.Element
-  prevArrow: JSX.Element
-}) => JSX.Element
+// const Slider = SliderClass as unknown as (props: {
```
Comment (Emmanueldmlr) [inline src/components/Elements/Carousel/Carousel.tsx:9]: Remove the comment.

File: src/components/Elements/Carousel/Carousel.tsx [hunk 2]
```diff
@@ -47,14 +56,60 @@ interface CarouselProps {
...
+
+  const fullSettings: Settings = {
+    ...settings,
+    // rome-ignore lint/correctness/noUnusedVariables: <explanation>
+    beforeChange: (current, next) => setActiveSlide(next),
+    afterChange: (current) => setActiveSlide2(current),
+  }
+  const sliderRef = useRef<SliderClass>(null)
+  const circles = [0, 1, 2, 3, 4]
+  const handleClick = (index: number) => {
```
Comment (Emmanueldmlr) [inline src/components/Elements/Carousel/Carousel.tsx:72]: Looking so complex, can you rewrite this?

File: src/components/Elements/Carousel/Carousel.tsx [hunk 3]
```diff
@@ -47,14 +56,60 @@ interface CarouselProps {
...
-    nextArrow={<ArrowBtn top={topArrow} ArrowIcon={FaChevronRight} isNext />}
-    prevArrow={<ArrowBtn top={topArrow} ArrowIcon={FaChevronLeft} />}
-  >
-    {children}
-  </Slider>
-)
+const Carousel = ({ settings, children, topArrow }: CarouselProps) => {
+  const [activeSlide, setActiveSlide] = useState(0)
+  // rome-ignore lint/correctness/noUnusedVariables: <explanation>
+  const [activeSlide2, setActiveSlide2] = useState(0)
```
Comment (kesar) [inline src/components/Elements/Carousel/Carousel.tsx:62]: https://docs.rome.tools/lint/rules/nounusedvariables/

use _ and avoid cutting corners w rome-ignores

**Comment by Royal-lobster** [comment]
Its acting weird after reaching to end and looping back to first card

https://user-images.githubusercontent.com/52039218/234218669-ca137ea8-67a8-433d-8888-85144c2821a5.mp4

**Comment by OleanjiKingCode** [comment]
it was meant to replicate Airbnb's own

https://user-images.githubusercontent.com/75235148/234496179-a6fb7b8d-6efb-4a93-aac9-c264aaf482d9.mp4

**Comment by Softdev1** [comment]
I would suggest that the arrows be included on the images so that users understand that there are still some featured wikis remaining

**Comment by Royal-lobster** [comment]
> it was meant to replicate Airbnb's own
> 
Ahh ok, but i think air bnb is having a animation which indicates the slider moving to next page. with it out its looking buggy imo.

**Review by Royal-lobster — APPROVED** [review]

**Comment by Adebesin-Cell** [comment]
https://user-images.githubusercontent.com/66301634/234603453-4c25a3cb-2416-4efc-af9e-2b6bdc87b039.mov

When the contents are longer than 5, the indicator isn't showing up. I think you can think up each slider item having dots since we have five elements rendered, then we give that one-page i.e. 5 elements per page, then after the fifth one we move to the next page with the elements inside.

**Comment by OleanjiKingCode** [comment]
it works normally 

https://user-images.githubusercontent.com/75235148/234606508-315aae49-1d63-4c7c-a589-108555752353.mp4

@Adebesin-Cell

**Review by Adebesin-Cell — APPROVED** [review]
Great work 👏

**Pull request merged**